mfs/dir.go

```diff
@@ -339,15 +339,24 @@ func (d *Directory) AddChild(name string, nd *dag.Node) error {
 
 	d.modTime = time.Now()
 
-	if len(nd.Links) == 0 {
+	pbn, err := ft.FromBytes(nd.Data)
+	if err != nil {
+		log.Error("IPNS pointer was not unixfs node")
```

Comment:
this should be a warning, we're going to see it anyways as it gets passed back up